app/src/main/java/org/schabi/newpipe/util/ListHelper.java

```diff
@@ -110,7 +110,8 @@ private static String computeDefaultResolution(Context context, int key, int val
                 : context.getString(R.string.best_resolution_key);
 
         String maxResolution = getResolutionLimit(context);
-        if (maxResolution != null && compareVideoStreamResolution(maxResolution, resolution) < 1){
+        if (maxResolution != null && (resolution.equals("best_resolution")
```

Comment:
Can you please show me, where the "best_resolution" comes from?
I do not have time to look it up and you already managed to find it once :)

Reply:
```java
System.out.println(preferences.getString(context.getString(R.string.default_resolution_key),
                                         context.getString(R.string.default_resolution_value)));
```

This line prints out `best_resolution` when Default resolution is set to Best resolution in settings.

Same stuff is happening here:
https://github.com/TeamNewPipe/NewPipe/blob/bb00a56de671d5d64b6f3884998b03410746ca2f/app/src/main/java/org/schabi/newpipe/util/ListHelper.java#L109

The `best_resolution` string seems to be arising from here:
https://github.com/TeamNewPipe/NewPipe/blob/fba0a8036bdfe046d2e10d3aa7011fc08592cb25/app/src/main/res/values/settings_keys.xml#L52

Reply:
Could be that i introduced this line, brcause sometimes the desired resolution is not available, and on different services not even possible. So you should select a resolution closest to the desired one.

Does that sound logic?

Reply:
> Does that sound logic?

Yes. 

What I wanted to note is that using `"best_resolution"` as String is not a good idea. Please use `context.getString(R.string.best_resolution_key)` instead. Otherwise, this bug might occur when we change the value of `R.string.best_resolution_key`.

Reply:
Yep, that's much better!